OSDOCS-3867: add references to throughput issues doc (#48331)
openshift/openshift-docs: modules/nw-throughput-troubleshoot.adoc

// Module filename: nw-throughput-troubleshoot.adoc
// Module included in the following assemblies:
// * networking/routes/route-configuration.adoc

:_content-type: CONCEPT
[id="nw-throughput-troubleshoot_{context}"]
= Throughput issue troubleshooting methods

> Sometimes applications deployed through {product-title} can cause network throughput issues such as unusually high latency between specific services.

Reviewer: ```suggestion
Sometimes applications deployed by using {product-title} can cause network throughput issues such as unusually high latency between specific services.
```
re: [ISG](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#t) guidance on _through_.

Reviewer: Forgot to add the note about the comma before _such as_

Use the following methods to analyze performance issues if pod logs do not reveal any cause of the problem:

> * Use a packet analyzer, such as ping or tcpdump to analyze traffic between a pod and its node.

Reviewer: ```suggestion
* Use a packet analyzer, such as `ping` or `tcpdump` to analyze traffic between a pod and its node.
```
A quick search of the docs shows `tcpdump` as the common usage with literals. Also, since ping could be used as a verb or as a command, I think it helps to differentiate.

> * Use a packet analyzer, such as ping or link:http://www.tcpdump.org/[tcpdump] to analyze traffic between a pod and its node.

Reviewer 2: Few suggestions the statement:
- Capturing tcpdump inside a pod and the on the node has different step and the implementation can differ specifically with dealing with pod level capture, I believe should include the general method such as article https://access.redhat.com/solutions/4569211 for a reference so the the reader will have better understanding of it. 
- The same applies to capturing tcpdump from the nodes, we have a good article : https://access.redhat.com/solutions/5074041 for it to be referenced. 
- For iperf testing https://access.redhat.com/solutions/6129701 will be a better reference to be added

Author: excellent, thanks for the feedback Arvind

Author: @quarterpin i've implemented your feedback, could you PTAL again for ack? Thank you!

+

> For example, link:https://access.redhat.com/solutions/4569211[run the tcpdump tool on each pod] while reproducing the behavior that led to the issue. Review the captures on both sides to compare send and receive timestamps to analyze the latency of traffic to and from a pod. Latency can occur in {product-title} if a node interface is overloaded with traffic from other pods, storage devices, or the data plane.

Reviewer: ```suggestion
For example, link:https://access.redhat.com/solutions/4569211[run the `tcpdump` tool on each pod] while reproducing the behavior that led to the issue. Review the captures on both sides to compare send and receive timestamps to analyze the latency of traffic to and from a pod. Latency can occur in {product-title} if a node interface is overloaded with traffic from other pods, storage devices, or the data plane.
```

+
[source,terminal]
----
$ tcpdump -s 0 -i any -w /tmp/dump.pcap host <podip 1> && host <podip 2> <1>
----
+
<1> `podip` is the IP address for the pod. Run the `oc get pod <pod_name> -o wide` command to get the IP address of a pod.
+
tcpdump generates a file at `/tmp/dump.pcap` containing all traffic between these two pods. Ideally, run the analyzer shortly before the issue is reproduced and stop the analyzer shortly after the issue is finished reproducing to minimize the size of the file. You can also link:https://access.redhat.com/solutions/5074041[run a packet analyzer between the nodes] (eliminating the SDN from the equation) with:
+
[source,terminal]
----
$ tcpdump -s 0 -i any -w /tmp/dump.pcap port 4789
----

> 

Reviewer: CIT: I think that you could make this into a series of short procedures interlaced with conceptual and reference materials. Check out the [OSDK Scorecard docs ](https://docs.openshift.com/container-platform/4.10/operators/operator_sdk/osdk-scorecard.html) for a potential model (though they aren't exactly troubleshooting docs).

Also, these docs rely on KCS articles, so if you add prerequisites and context from those, you might have enough to flesh these out a bit.

> * Use a bandwidth measuring tool, such as link:https://access.redhat.com/solutions/6129701[iperf], to measure streaming throughput and UDP throughput. Run the tool from the pods first, then from the nodes, to locate any bottlenecks.

Reviewer: ```suggestion
* Use a bandwidth measuring tool, such as link:https://access.redhat.com/solutions/6129701[`iperf`], to measure streaming throughput and UDP throughput. Locate any bottlenecks by running the tool from the pods first, and then running it from the nodes.
```
Does UDP need to be expanded?

Author: It is mostly kept abbreviated in the docs, I think in many cases UDP is more recognizable than what it's abbreviating 😄 (user diagram protocol). I think customers in this part of the docs are familiar with networking principles e.g. TCP/UDP/IP

ifdef::openshift-enterprise,openshift-webscale[]

> ** For information on installing and using iperf, see this link:https://access.redhat.com/solutions/33103[Red Hat Solution].

Reviewer: ```suggestion
** For information on installing and using `iperf`, see this link:https://access.redhat.com/solutions/33103[Red Hat Solution].
```

endif::openshift-enterprise,openshift-webscale[]

> * In some cases, the cluster may mark the node with the router pod as unhealthy due to latency issues. Use worker latency profiles to adjust the frequency that the cluster waits for a status update from the node before taking action.

Reviewer: Could we expand this with a little bit of starter information how to use a worker latency profile like the steps above? Again, this could be a short procedure. If digital rot and procedure change is a concern, we could reference customers to a place in the docs where this is described.

Reviewer: CIT suggestion ^

Author: yeah for sure. The additional resources section in the assembly links to that information, but I agree a short procedure on how to enable worker profiles would be really useful here

> * Configure the `spec.nodePlacement` field in the Ingress Controller to control the placement of the router pod, in the case that your cluster has designated lower-latency and higher-latency nodes.

Reviewer: ```suggestion
* If your cluster has designated lower-latency and higher-latency nodes, configure the `spec.nodePlacement` field in the Ingress Controller to control the placement of the router pod.
```
Is there an example YAML that you can excerpt here?
